### PR TITLE
Fix omeganuh2

### PR DESCRIPTION
**Pull request by c-d-leonard**
Fix the issues described in Issue #780 where ccl.Omeganuh2 has unexpected behaviour.

### CODE REVIEW

**Comment by c-d-leonard** [comment]
@slosar could you take a look at this? it fixes the Omeganuh2 dodginess you pointed out.

In particular, the C-level function was doing a thing where if the first quantity in the mass vector was ~0, it was using the massless case and only returning that. This was a buggy holdover from old code. The Omeganuh2 function is intended for the massive neutrino case, and in the case someone would call it with a zero or very low mass neutrino (e.g. by accident) you still get the right answer if you compute the full integral as for the massive case, so I've just had it do that.

I also fixed the issue where it wasn't accepting a float; that was a weird hard-coding thing in the ccl_neutrions.i file, should be fine now.

File: src/ccl_neutrinos.c [hunk 1]
```diff
@@ -170,14 +170,7 @@ double ccl_Omeganuh2 (double a, int N_nu_mass, double* mnu, double T_CMB, int* s
 
   Tnu = T_CMB*pow(4./11.,1./3.);
   a4 = a*a*a*a;
-  // Check if mnu=0. We assume that in the massless case mnu is a
-  // pointer to a single element and that element is 0. This should in principle never be called.
-  if (mnu[0] < 0.00017) {  // Limit taken from Lesgourges et al. 2012
```
Comment (slosar) [inline src/ccl_neutrinos.c:175]: But did you actually have to remove this line? It does seem to be sensible to revert to the default calculation ofr low enough mass.

Comment (c-d-leonard) [reply src/ccl_neutrinos.c:175]: Fair point. I was being a bit lazy by removing it as it almost never should get hit (only for the case where we have to post-hoc fudge equal neutrino masses for cosmic emu, I think). But I've reintroduced that case now (in the below loop so we make sure to check them all and not just the first as was causing some of the original weird behaviour).

Comment (c-d-leonard) [reply src/ccl_neutrinos.c:175]: Thoughts @slosar? If you're happy could you approve?

**Pull request merged**